datafusion/core/src/execution/session_state.rs

```diff
@@ -1922,4 +1929,32 @@ mod tests {
         assert!(new_state.catalog_list().catalog(&default_catalog).is_none());
         Ok(())
     }
+
+    #[test]
+    fn test_session_state_with_optimizer_rules() {
+        // test building sessions with supplied rules
```

Comment:
nit
```suggestion
        struct DummyRule {}

        impl OptimizerRule for DummyRule {
            fn name(&self) -> &str {
                "dummy_rule"
            }
        }
        // test building sessions with supplied rules
```

Reply:
@Weijun-H - Thanks for having a look! Applied your feedback.

https://github.com/apache/datafusion/pull/12379/commits/bba3fe52b022c7a01e524cc138ee57d2f957f274